src/types/value/link.ts

```diff
@@ -15,6 +15,16 @@ export const LinkType = {
 	Web: "Web",
 } as const
 
+/**
+ * Additional optional data for a link field.
+ *
+ * @typeParam Link - Link field.
+ */
+export type WithLinkAdditionalData<Link> = Link & {
```

Comment:
what about
```suggestion
export type WithAdditionalLinkAttributes<Link> = Link & {
```
another option: WithAdditionalLinkProps